dev/bots/analyze_framework_code.dart

```diff
@@ -0,0 +1,421 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:analyzer/dart/analysis/analysis_context.dart';
+import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/analysis/session.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:analyzer/source/line_info.dart';
+import 'package:collection/collection.dart';
+import 'package:path/path.dart' as path;
+
+import 'utils.dart';
+
+Future<void> runVerifiersInResolvedDirectory(String workingDirectory, List<ResolvedUnitVerifier> verifiers) async {
```

Comment:
everything public in this file would benefit from dartdoc documentation so other people can easily write custom verifiers and understand what the existing ones do.

Reply:
(If we end up putting these in a sub-directory, it may be worthwhile to enable that lint requiring doc comments for that sub directory)

Reply:
I tried adding an analysis_options file to the sub directory but that didn't work. I guess that's because the directory doesn't contain a package? [This page](https://dart.dev/tools/analysis#the-analysis-options-file) says 

> If the analyzer can’t find an analysis options file at the package root, it walks up the directory tree, looking for one. If no file is available, the analyzer defaults to standard checks.

So I guess the rules are not per directory but per package?